test/unit/helpers/wsApi.js

```diff
@@ -2,125 +2,195 @@
 
 var config = require('../../../config.json');
 
+var async = require('async');
 var chai = require('chai');
 var expect = require('chai').expect;
 var express = require('express');
-var _  = require('lodash');
 var sinon = require('sinon');
 
 var wsApi = require('../../../helpers/wsApi');
+var failureCodes = require('../../../api/ws/rpc/failureCodes');
 
 var System = require('../../../modules/system');
 
 describe('handshake', function () {
 
-	var system, handshake, validRequest;
-
-	before(function (done) {
-
-		sinon.stub(System.prototype, 'getNethash').returns(config.nethash);
-		sinon.stub(System.prototype, 'getMinVersion').returns(config.version);
-
+	var system;
+	var handshake;
+	var validConfig;
+	var validHeaders;
+	var minVersion = '1.0.0';
+	var nonStrings = [{}, [], 1, 0.1, NaN, true];
+	var nonNumbers = [{}, [], 'A', '1', NaN, true];
+
+	beforeEach(function (done) {
+		validConfig = {
+			config: {
+				version: config.version,
+				minVersion: minVersion,
+				nethash: config.nethash,
+				nonce: 'ABCDEF0123456789'
+			}
+		};
 		new System(function (err, __system) {
 			system = __system;
 			handshake = wsApi.middleware.Handshake(system);
 			done(err);
-		}, {config: {
-			version: config.version,
-			minVersion: config.minVersion,
-			nethash: config.nethash,
-			nonce: 'EXAMPLE_NONCE'
-		}});
+		}, validConfig);
 
-		validRequest = {
-			ip: '0.0.0.0',
-			port: 4000,
+		validHeaders = {
+			port: 5000,
 			nethash: config.nethash,
-			version: config.version,
-			nonce: 'PEER_NONCE'
+			version: minVersion,
+			nonce: '0123456789ABCDEF',
+			height: 1
 		};
 	});
 
-	it('should accept handshake when valid request passed', function (done) {
-
-		handshake(validRequest, function (err, data) {
+	it('should accept handshake for valid headers', function (done) {
```

Comment:
Nice. I would just create a context block for the errors. It will be more readable.